nilearn/plotting/img_plotting.py

```diff
@@ -2409,3 +2412,262 @@ def plot_img_comparison(
+    ax3.hist(
+        mean,
+        bins=gridsize[1],
+        range=lims[0:2],
+        histtype="stepfilled",
+        orientation="vertical",
+        color="gray",
+    )
+    ax3.set_xlim(lims[0:2])
+    ax3.invert_yaxis()
+    ax3.set_xlabel(f"Average :  mean({ref_label}, {src_label}")
+
+    ax4 = figure.add_subplot(gs[:-1, 5])
+    ax4.set_aspect(20)
+    pos1 = ax4.get_position()
+    ax4.set_position([pos1.x0 - 0.025, pos1.y0, pos1.width, pos1.height])
+
+    cb = figure.colorbar(hb, cax=ax4)
+    cb.set_label("log10(N)")
+
+    if output_file is not None:
+        figure.savefig(output_file)
+        plt.close(figure)
+        figure = None
+
+    return figure
+
+
+def _bland_altman_values(ref_img, src_img, masker=None):
+    """Return voxel wise mean and differences between 2 images."""
+    if isinstance(ref_img, (str, Path, Nifti1Image)) and isinstance(
+        src_img, (str, Path, Nifti1Image)
+    ):
+        image_type = "volume"
+        ref_img = check_niimg_3d(ref_img)
+        src_img = check_niimg_3d(src_img)
+
+    elif isinstance(ref_img, (SurfaceImage)) and isinstance(
+        src_img, (SurfaceImage)
+    ):
+        image_type = "surface"
+        ref_img.data._check_ndims(1)
+        src_img.data._check_ndims(1)
+        check_same_n_vertices(ref_img.mesh, src_img.mesh)
+
+    else:
+        raise TypeError(
+            "'ref_img' and 'src_img' "
+            "must both be Niimg-like or SurfaceImage.\n"
+            f"Got {type(src_img)=} and {type(ref_img)=}."
+        )
+
+    if masker is not None:
+        if image_type == "volume" and not isinstance(
+            masker, (NiftiMasker, Nifti1Image, str, Path)
+        ):
+            raise TypeError(
+                "'masker' must be NiftiMasker or Niimg-Like "
+                "for volume based images.\n"
+                f"Got {type(masker)}"
+            )
+
+        elif image_type == "surface" and not isinstance(
+            masker, (SurfaceMasker, SurfaceImage)
+        ):
+            raise TypeError(
+                "'masker' must be SurfaceMasker or SurfaceImage "
+                "for surface based images.\n"
+                f"Got {type(masker)}"
+            )
+
+        if image_type == "volume" and isinstance(
+            masker, (Nifti1Image, str, Path)
+        ):
+            masker = NiftiMasker(
+                mask_img=masker,
+                target_affine=ref_img.affine,
+                target_shape=ref_img.shape,
+            )
+
+        elif image_type == "surface" and isinstance(masker, SurfaceImage):
+            check_same_n_vertices(ref_img.mesh, masker.mesh)
+            masker = SurfaceMasker(
+                mask_img=masker,
+            )
+
+    elif image_type == "volume":
+        masker = NiftiMasker(
+            target_affine=ref_img.affine,
+            target_shape=ref_img.shape,
+        )
+
+    elif image_type == "surface":
+        masker = SurfaceMasker()
```

Comment:
Could the code be easier to read if these if condition statements are placed inside the above block where relevant?